### ISSUE TITLE
XHCI is 0xDEAD on FreeBSD

### ISSUE THREAD

**Issue by myfreeweb** [pftf/edk2-platforms#9]
(RPi4 4GB) trying 1.2 firmware on FreeBSD -CURRENT, xhci doesn't work: reading from `0x600000000` memory results in stuff like:

- "cap length" `0xad`
- "xhci version" `0xdead`
- "runtime offset" `0xdeaddea0`
- "door offset" `0xdeaddeac`

What could that be?..

**Comment by pbatard** [comment]
If you want to report an issue with RPi4 firmware releases, please report them in the [RPi4 firmware repository](https://github.com/pftf/RPi4/releases).

Also, you haven't indicated whether XHCI works with firmware 1.0 or 1.1. What's the last version of the firmware that worked?

**Comment by myfreeweb** [comment]
Haven't tried previous versions before.. Same behavior on 1.1.

**Comment by pbatard** [comment]
Okay, so this isn't a regression.

You do understand that this firmware is very experimental  and almost __no single__ OS is expect to work fully at this stage, right?

**Comment by myfreeweb** [comment]
Of course. I'm not asking for you to fix everything, I'm just wondering if you have anything that might help with debugging, if anyone has seen this behavior on any other OS, something like that?

**Comment by andreiw** [comment]
Can you jump onto the Discord channel? We're actively discussing this (with one of your FreeBSD colleagues)

Either _INI is not being run for the device or your accessing garbage.

Good:
![image](https://user-images.githubusercontent.com/732523/81441832-15504680-9138-11ea-93fe-583f32bcdc1b.png)

Bad:
![image](https://user-images.githubusercontent.com/732523/81441839-1a14fa80-9138-11ea-84a5-37c272de2b72.png)
